### ISSUE TITLE
Ignore `mutable-argument-default (B006)` in `.pyi` stubs

### ISSUE THREAD

**Issue by jorenham** [astral-sh/ruff#14026]
[Mutable argument defaults](https://docs.astral.sh/ruff/rules/mutable-argument-default/) can only be a problem at runtime. But at the moment, `B006` will also be reported in `.pyi` stub files, even though it is certain that mutations can never occur there.

For example, consider the following function in [`scipy-stubs/stats/_mstats_basic.pyi:578`](https://github.com/jorenham/scipy-stubs/blob/5a2603e9fc3c37e9355773da31d294ac688d699d/scipy-stubs/stats/_mstats_basic.pyi#L578-L585):

```pyi
def mquantiles(
    a: _ArrayLikeFloat_co,
    prob: _ArrayLikeFloat_co = [0.25, 0.5, 0.75],
    alphap: AnyReal = 0.4,
    betap: AnyReal = 0.4,
    axis: CanIndex | None = None,
    limit: tuple[AnyReal, AnyReal] | tuple[()] = (),
) -> _MArrayND: ...
```

For the `prob` parameter the following error is reported:

```
scipy-stubs/stats/_mstats_basic.pyi:580:32: B006 Do not use mutable data structures for argument defaults
```

I think that "mutable" defaults should be allowed in cases like these. Having explicit parameter defaults that match the runtime can be seen as a form of documentation. This is especially useful when the runtime sources aren't available, which is quite common in this [`scipy-stubs`](https://github.com/jorenham/scipy-stubs) case.

**Comment by AlexWaygood** [comment]
Yeah, your argument makes sense. You want to emulate the runtime in a stub file; whether or not the runtime _should_ be using mutable defaults or not, you'll want to accurately represent that in the stub.
